spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -1093,6 +1093,69 @@ When native encoding is used, it is the responsibility of the consumer to use ap
 Also, when native encoding/decoding is used the `headerMode` property is ignored and headers will not be embedded into the message.
 +
 Default: `false`.
+[[dynamicdestination]]
+== Binding producer to destinations dynamically at runtime
+
+It is useful to have the Spring Cloud Stream 'outbound' applications sending messages to a 'destination' that is determined dynamically as the messages are processed at runtime.
+In these cases, the destination names to the broker (for example: exchange name in case Rabbit or topic name in case of Kafka) may or may not be available beforehand.
+Spring Cloud Stream applications can make use 'BinderAwareChannelResolver' which is available to the application when '@EnableBinding' is annotated to the application.
+
+The property 'spring.cloud.stream.dynamicDestinations' can be set if the dynamic destinations are known and the application needs to bind to this list of dynamic destinations **only**. If this property is set to empty(which is by default empty), then 'BinderAwareChannelResolver' takes care of binding to all the resolved dynamic destinations.
+
+
+For instance, the below example application registers a REST 'controller' with a request mapping on the path '/'. It also has 'router' that resolves the dynamic destinations using the SpEL expression evaluated against the message payload.
+
+[source,java]
```

Comment:
Would it make sense to provide a simpler example based on the direct use of `BinderAwareChannelResolver.resolveDestination()` *and* use the router example as an illustration of how the component can be used in a more advanced scenario?